pkg/distributor/distributor.go

```diff
@@ -519,6 +530,12 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 				structuredMetadata := logproto.FromLabelAdaptersToLabels(entry.StructuredMetadata)
 				if shouldDiscoverLevels {
```

Comment:
Why do we do this only for should DiscoverLevels ?

Reply:
unintended, moved the checks to before the `shouldDiscoverLevels` block